### PR TITLE
docs: add api docs for apiRequestTime and apiRequestSize

### PR DESCRIPTION
**Pull request by watson**
Changes for intake API v2

### CODE REVIEW

**Comment by watson** [comment]
The CI failures are because of the npm bug with the github link in the dependencies

File: docs/performance-tuning.asciidoc [hunk 1]
```diff
@@ -35,35 +35,6 @@ require('elastic-apm-node').start({
 })
 ----
 
-[float]
-[[performance-transaction-queue]]
-=== Transaction Queue
```
[the remark this answers is not in the document]
Comment (Qard) [reply docs/performance-tuning.asciidoc]: We should probably replace it with a new section describing how `apiRequestTime` and `apiRequestSize` will impact the performance.

[the remark this answers is not in the document]
Comment (watson) [reply docs/performance-tuning.asciidoc]: Updated

File: docs/performance-tuning.asciidoc [hunk 2]
```diff
@@ -36,13 +36,55 @@ require('elastic-apm-node').start({
 ----
 
 [float]
+[[performance-apm-server-communication]]
+=== APM Server communication
+
+The agent uses a persistent outgoing HTTP request to stream data to the APM Server.
+To avoid issues with intermittent proxies and load balancers,
+the HTTP request is ended and a new one created at regular intervals or when the size of the reqeust becomes too big.
```
Comment (watson) [inline docs/performance-tuning.asciidoc]: s/reqeust/request/

File: docs/performance-tuning.asciidoc [hunk 3]
```diff
@@ -36,13 +36,55 @@ require('elastic-apm-node').start({
...
+as headers and metadata doesn't need to be re-sent that often.
+
+[[performance-api-request-size]]
+==== Max HTTP request size
+
+By default an HTTP request to the APM Server is ended after 1 MiB of gzip compressed data have been written to the body.
+Using the <<api-request-size,`apiRequestSize`>> config option,
+this time limit can be modified.
+
+Lowering the size limit might be nessecary if dealing with very agressive proxies,
```
Comment (Qard) [inline docs/performance-tuning.asciidoc]: necessary*

File: docs/performance-tuning.asciidoc (hunk 2 above)
Comment (watson) [reply docs/performance-tuning.asciidoc]: Fixed

File: docs/performance-tuning.asciidoc (hunk 3 above)
Comment (watson) [reply docs/performance-tuning.asciidoc]: Fixed

**Comment by Qard** [comment]
There seems to be an asciidoc error. 🤔 

```
WARNING: performance-tuning.asciidoc: line 54: section title out of sequence: expected level 2, got level 3
```

**Pull request merged**